src/tests/multihost/alltests/test_backtrace.py

```diff
@@ -0,0 +1,167 @@
+
+:requirement: Poor Man's Backtrace
+:casecomponent: sssd
+:subsystemteam: sst_idm_sssd
+:upstream: yes
+:bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1949149
+"""
+
+from __future__ import print_function
+import re
+import time
+import pytest
+from sssd.testlib.common.utils import sssdTools
+from sssd.testlib.common.expect import pexpect_ssh
+from constants import ds_instance_name
+
+
+@pytest.mark.usefixtures('setup_sssd', 'create_posix_usersgroups')
+@pytest.mark.backtrace
+@pytest.mark.tier1_2
+class TestPoorManBacktrace(object):
+    """ Check sssd backtrace feature """
+    def test_0001_bz2021196(self, multihost, backupsssdconf):
+        """
+        :title: avoid duplicate backtraces
+        :id: d4d8a0a0-ab90-4c8f-8087-95dc7ad3f3ae
+        :customerscenario: true
+        :bugzilla:
+          https://bugzilla.redhat.com/show_bug.cgi?id=2021196
+          https://bugzilla.redhat.com/show_bug.cgi?id=2021499
+        """
+        hostname = multihost.master[0].sys_hostname
+        bad_ldap_uri = f"ldaps://typo.{hostname}"
+        tools = sssdTools(multihost.client[0])
+        domain_params = {'ldap_uri': bad_ldap_uri}
+        tools.sssd_conf(f'domain/{ds_instance_name}', domain_params)
+        tools.clear_sssd_cache()
+        logfile = '/var/log/sssd/sssd_nss.log'
+        cmd = f'getent passwd fakeuser@{ds_instance_name}'
+        multihost.client[0].run_command(cmd, raiseonerr=False)
+        msg = 'BACKTRACE DUMP ENDS HERE'
+        msg2 = '... skipping repetitive backtrace ...'
+        pattern = re.compile(fr'{msg}')
+        pattern2 = re.compile(fr'{msg2}')
+        log_str1 = multihost.client[0].get_file_contents(logfile). \
+            decode('utf-8')
+        multihost.client[0].run_command(f'> {logfile}')
+        multihost.client[0].run_command(cmd, raiseonerr=False)
+        time.sleep(2)
+        log_str2 = multihost.client[0].get_file_contents(logfile). \
+            decode('utf-8')
+        # Check the backtrace is dumped first time and no backtrace is skipped
+        assert pattern.search(log_str1) and not pattern2.search(log_str1)
+        # Check there is no new backtrace with the same issue and repeative
+        # backtrace is skipped
+        assert pattern2.search(log_str2) and not pattern.search(log_str2)
+
+    def test_0001_bz1949149(self, multihost, backupsssdconf):
+        """
+        :title: backtrace is disabled if debug level >= 9
+        :id: 50f2d501-3296-4229-86a0-b81844381637
+        """
+        section = f"domain/{ds_instance_name}"
+        param = {'debug_level': '9'}
+        serv_list = ['sssd', section, 'nss', 'pam']
+        tools = sssdTools(multihost.client[0])
+        for serv in serv_list:
+            tools.sssd_conf(serv, param)
+        tools.clear_sssd_cache()
+        cmd_kill = 'kill -SIGUSR2 $(pidof sssd)'
+        multihost.client[0].run_command(cmd_kill, raiseonerr=False)
+        cmd = f'getent passwd fakeuser@{ds_instance_name}'
```

Comment:
But lookup of unknown user shouldn't trigger a backtrace (doesn't matter what debug level set).
Does it? If it does we probably need to change debug level for this case...

Reply:
You are right, It does not trigger a backtrace. It is now used to generate a backtrace for connection failure when lookup is initiated.